src/docGenerator.ts

```diff
@@ -0,0 +1,146 @@
+import fs from "fs"
```

Comment:
in this file there are too many concerns
file system operations, json handling and more,
please split the different functionalities for clarity